sunkit_instruments/suvi/suvi.py

```diff
@@ -0,0 +1,1122 @@
+    gain_file = os.path.join(path_to_files, 'SUVI_'+flight_model[spacecraft]+'_gain.txt')
+
+    # Get effective areas
+    eff_area = numpy.loadtxt(eff_area_file, skiprows=12)
+    wave = eff_area[:,0] * u.Angstrom
+    if filter_setup[wavelength_channel][exposure_type]['FW2'] == 'open':
+        effective_area = eff_area[:,1] * u.cm * u.cm
+    else:
+        effective_area = eff_area[:,2] * u.cm * u.cm
+    
+    # Get gain
+    gain_table = numpy.loadtxt(gain_file, skiprows=7)
+    temp_x = gain_table[:,0]
+    gain_y = gain_table[:,1]
+    gain_vs_temp = interpolate.interp1d(temp_x, gain_y)
+    gain = gain_vs_temp(ccd_temperature)
+
+    geometric_area = 19.362316 * u.cm * u.cm
+    solid_angle = ((2.5/3600. * (numpy.pi/180.))**2.) * u.sr
+    master_e_per_phot = ((6.626068e-34 * (u.J/u.Hz)) * (2.99792458e8 * (u.m/u.s)))/(wave.to(u.m) * ((u.eV.to(u.J, 3.65)) * u.J))    
+
+    response = effective_area * (master_e_per_phot/gain)
+
+    response_info = {'wavelength': wave, 'effective_area': effective_area * (u.ct/u.ph), 'response': response,
+                     'wavelength_channel': wavelength_channel, 'spacecraft': 'GOES-'+str(spacecraft),
+                     'ccd_temperature': ccd_temperature * u.deg_C, 'exposure_type': exposure_type, 
+                     'flight_model': flight_model[spacecraft], 'gain': numpy.float64(gain),
+                     'solid_angle': solid_angle, 'geometric_area': geometric_area, 
+                     'filter_setup': filter_setup[wavelength_channel][exposure_type]}
+
+    return response_info
+
+
+def plot_thematic_map(input_filename, timestamp=True, legend=True, figsize=(10,10)):
+    """
+    Read a SUVI L2 Thematic Map FITS file and plot it.
+
+    .. note::
+        SUVI L2 Thematic Maps are recognized by pattern matching in the 
+        filenames, i.e. it must contain "-l2-thmap". If this pattern is
+        not found in the filename, it will not be recognized.
+
+    .. note::
+       GOES-16 L2 Thematic Maps are available
+       `here. <https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes16/l2/data/suvi-l2-thmap/>`__
+
+    Parameters
+    ----------
+    input_filename: `str`
+        File to read.
+
+    timestamp: `bool`, optional. Default: True. 
+        If True, plots the timestamp of the observation. 
+
+    legend: `bool`, optional. Default: True. 
+        If True, plots a legend with the different classes.
+
+    figsize: `tuple`, optional. Default: (10, 10).
+        The size of the figure. Should not be smaller than 
+        (7, 7) to look decent.
+    """
+
+    if '-l2-thmap' in os.path.basename(input_filename):
+        hdu = fits.open(input_filename)
+        thmap_data = hdu[0].data
+        if timestamp:
+            time_st = hdu[0].header['DATE-OBS'][0:19]
+        hdu.close()
+        
+        SOLAR_CLASSES = [('unlabeled', 0),
```

Comment:
Can anything that's fixed like these variables be moved to the top of the file?

Reply:
Those are only used in that function, no other one - should I still do that?

Reply:
Yes please.